camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlInputFormat.java

```diff
@@ -86,6 +87,11 @@
   public static final int NUM_TRIES_PARTITION_METADATA = 3;
   public static final int NUM_TRIES_FETCH_FROM_LEADER = 3;
   public static final int NUM_TRIES_TOPIC_METADATA = 3;
+  // Camus will alert if the number-of-offsets-to-process is
+  // FALLING_BEHIND_ALERT_OFFSETS_MULTIPLER times the number-of-offsets-left-in-retention ....
+  public static final int FALLING_BEHIND_ALERT_OFFSETS_MULTIPLER = 3;
+  /// ...but only if the absolute number of offsets to process is greater than FALLING_BEHIND_ALERT_OFFSETS_THRESHOLD
+  public static final int FALLING_BEHIND_ALERT_OFFSETS_THRESHOLD = 1000000;
```

Comment:
too much?

Reply:
I think the reason the original logic uses a multiplier is because an absolute threshold is topic-dependent. With too small a number it's too chatty, with too large -- it will miss a legitimate case of partition being stuck. I've actually figured out how to report on stuck offsets from Splunk, _I think_. Maybe the best solution in the meantime is to not mess with the condition but just to report to datadog the same thing that was reported via email before (just so we have alerts in one place) and we can periodically (e.g. twice a day) check up on those alerts until we convince ourselves that the spunk alert works?

Reply:
Adding the absolute check in _addition to_ the multiplier would just remove most of the noise, it's pragmatic, I think. But I agree that once we have the splunk alerts, we'll be able to be less stringent on this particular data point. So, I'll remove this and rely on the splunk alerts for stuck partitions.